internal/controller/controllers/cluster_controller.go

```diff
@@ -121,17 +157,38 @@ func (r *ClusterReconciler) createNewCluster(
 	})
 	// TODO: handle specific errors, 5XX retry, 4XX update status with the error
 	if err != nil {
-		cluster.Status.Error = err.Error()
-		if err = r.Update(ctx, cluster); err != nil {
-			r.Log.WithError(err).Error("failed to update error status")
-		}
-		return ctrl.Result{}, errors.Wrap(err, "failed to create cluster")
+		return r.updateErrorStatus(ctx, cluster, err)
 	}
 
-	return r.updateStatus(ctx, cluster, c)
+	return r.updateState(ctx, cluster, c)
 }
 
-func (r *ClusterReconciler) updateStatus(ctx context.Context, cluster *adiiov1alpha1.Cluster, c *common.Cluster) (ctrl.Result, error) {
+func (r *ClusterReconciler) updateErrorStatus(ctx context.Context, cluster *adiiov1alpha1.Cluster, err error) (ctrl.Result, error) {
+	cluster.Status.Error = err.Error()
+	if err = r.Status().Update(ctx, cluster); err != nil {
+		r.Log.WithError(err).Error("failed to update error status")
+		return ctrl.Result{Requeue: true}, nil
+	}
+	r.Log.Infof("Updated cluster %s %s error status", cluster.Name, cluster.Namespace)
+	return ctrl.Result{RequeueAfter: defaultRequeueClusterDuration}, nil
+}
+
+func (r *ClusterReconciler) updateStateWithError(ctx context.Context, cluster *adiiov1alpha1.Cluster,
```

Comment:
not sure I understood what's the point of having `updateStateWithError` function and `updateErrorStatus` where the only difference is that here we also call `r.syncClusterState(cluster, c)` 
(hard to understand the difference from the name)
Is this intentional?

Reply:
You are right, i will try to use a single function.